plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/Init.php

```diff
@@ -29,34 +29,31 @@ public function __construct() {
 	 * @return array
 	 */
 	public static function get_extensions( $allowed_bundles = array() ) {
-		$bundles = array();
-		$specs   = self::get_specs();
+		$locale = get_user_locale();
 
-		foreach ( $specs as $spec ) {
-			$spec              = (object) $spec;
-			$bundle            = (array) $spec;
-			$bundle['plugins'] = array();
+		$specs   = self::get_specs();
+		$results = EvaluateExtension::evaluate_bundles( $specs, $allowed_bundles );
 
-			if ( ! empty( $allowed_bundles ) && ! in_array( $spec->key, $allowed_bundles, true ) ) {
-				continue;
+		$plugins = array_filter(
+			$results['bundles'],
+			function( $bundle ) {
+				return count( $bundle['plugins'] ) > 0;
 			}
+		);
 
-			foreach ( $spec->plugins as $plugin ) {
-				try {
-					$extension = EvaluateExtension::evaluate( (object) $plugin );
-					if ( ! property_exists( $extension, 'is_visible' ) || $extension->is_visible ) {
-						$bundle['plugins'][] = $extension;
-					}
-					// phpcs:ignore Generic.CodeAnalysis.EmptyStatement.DetectedCatch
-				} catch ( \Throwable $e ) {
-					// Ignore errors.
-				}
-			}
+		// When no plugins are visible, replace it with defaults and save for 3 hours.
+		if ( empty( $plugins ) ) {
```

Comment:
So when there's no errors, it should use the default behaviour (save it to transient for 7 days I think?)

Reply:
> This is because it's possible to have intentionally empty specs (no partners available) 🙏

@ilyasfoo Good point! But I wonder when or why would we need to have empty specs?  If it's empty, the payment task or marketing task will display empty content. 

I think it's also possible that even If it's empty and there are no errors, it might also be an unexpected behavior. In this case, falling back to default in this case might also help prevent that.

> So when there's no errors, it should use the default behaviour (save it to transient for 7 days I think?)

Yes!

Reply:
Good points, @chihsuan. I do agree it's probably not a realistic use case. Should be good then!